lib/util/BandUtil.js

```diff
@@ -75,39 +75,42 @@ export function resetAllBands(activity, diBands, bounds) {
 
 /**
  * Moves and resizes the participant bands of a choreography activity
- * according to the activity's bounds. Updates both the shapes as well
- * as the underlying DI objects.
+ * according to the activity's bounds or the changed band bounds.
+ * Additionally the label is repositioned by a redraw of the activity if necessary.
+ * Updates both the shapes as well as the underlying DI objects.
  *
  * @param {Object} activityShape shape object of the choreography activity
- * @param {Object} oldBounds old bounds of the choreography activity
- * @param {Object} newBounds new bounds of the choreography activity
+ * @param eventBus {EventBus} the event bus
  */
-export function resizeBands(activityShape, oldBounds, newBounds) {
+export function updateBands(activityShape, eventBus) {
+  const newBounds = activityShape.businessObject.di.bounds;
   let bandShapes = activityShape.bandShapes;
 
-  // all bands' widths needs to be adapted
+  // all bands' widths and height needs to be adapted, as well as x-position
   bandShapes.forEach(bandShape => {
     bandShape.x = newBounds.x;
     bandShape.width = newBounds.width;
     bandShape.diBand.bounds.x = newBounds.x;
     bandShape.diBand.bounds.width = newBounds.width;
+    bandShape.height = getBandHeight(bandShape.businessObject);
+    bandShape.diBand.bounds.height = bandShape.height;
   });
 
   let gapIndex = getBandGapIndex(bandShapes.length);
   let topShapes = bandShapes.slice(0, gapIndex);
   let bottomShapes = bandShapes.slice(gapIndex);
 
-  // move top bands up
-  let deltaTop = newBounds.y - oldBounds.y;
+  let offsetTop = 0;
   topShapes.forEach(shape => {
-    shape.y += deltaTop;
+    shape.y = newBounds.y + offsetTop;
     shape.diBand.bounds.y = shape.y;
+    offsetTop += shape.height;
   });
 
-  // move bottom bands down
-  let deltaBottom = (newBounds.height - oldBounds.height) + deltaTop;
-  bottomShapes.forEach(shape => {
-    shape.y += deltaBottom;
+  let offsetBottom = 0;
+  bottomShapes.slice().reverse().forEach(shape => {
```

Comment:
Superfluous `slice()`?

Reply:
No it is necessary, reverse is in place.